### PR TITLE
Add description for Detection

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Add docs for Detection

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/reference/object_detection.rst [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+The :class:`~flash.vision.ObjectDetector` is already pre-trained on `COCO train2017 <https://cocodataset.org/>`_, a dataset with `91 classes <https://cocodataset.org/#explore>`_ (123,287 images, 886,284 instances).
+
+
+Use the :class:`~flash.vision.ObjectDetector` pretrained model for inference on any image tensor or image path using :func:`~flash.vision.ObjectDetector.predict`:
+
+.. code-block:: python
+
+	from flash.vision import ObjectDetector
+
+	# Load the model
```
Comment (edenlightning) [inline docs/source/reference/object_detection.rst:29]:
```suggestion
	# 1. Load the model
```

File: docs/source/reference/object_detection.rst [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+	# 2. Perform inference on an image file
+	predictions = detector.predict("path/to/image.png")
+	print(predictions)
+
+Or on a random image tensor
+
+.. code-block:: python
+
+    # 2. Perform inference on a random image tensor
```
Comment (edenlightning) [inline docs/source/reference/object_detection.rst:40]:
```suggestion
    # Perform inference on a random image tensor
```

File: docs/source/reference/object_detection.rst [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+For more advanced inference options, see :ref:`predictions`.
+
+------
+
+**********
+Finetuning
+**********
+
+To tailor the object detector to your dataset, you would need to have it in `COCO Format <https://cocodataset.org/#format-data>`_, and then finetune the model.
```
Comment (edenlightning) [inline docs/source/reference/object_detection.rst:54]: Can you explain what this means briefly?

File: docs/source/reference/object_detection.rst [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+################
+Object Detection
+################
+
+********
+The task
+********
+
+Object Detection task helps to identify and locate objects in images.
```
Comment (edenlightning) [inline docs/source/reference/object_detection.rst:12]:
```suggestion
The object detection task identifies instances of objects of a certain class within an image.```

File: docs/source/reference/object_detection.rst (hunk 3 above)
Comment (kaushikb11) [reply docs/source/reference/object_detection.rst:54]: I think the link https://cocodataset.org/#format-data explains it well. We need metadata in this format for the images in the folder.

**Review by Borda — CHANGES_REQUESTED** [review]
just add it to index :]

File: docs/source/reference/object_detection.rst (hunk 3 above)
Comment (kaushikb11) [reply docs/source/reference/object_detection.rst:54]: @edenlightning

File: docs/source/reference/object_detection.rst [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+################
+Object Detection
+################
+
+********
+The task
+********
+
+The object detection task identifies instances of objects of a certain class within an image.```
```
Comment (Borda) [inline docs/source/reference/object_detection.rst:12]:
```suggestion
The object detection task identifies instances of objects of a certain class within an image.
```

File: docs/source/reference/object_detection.rst (hunk 3 above)
Comment (Borda) [reply docs/source/reference/object_detection.rst:54]: it would be useful to add a short example, no need for details but give user idea...

**Review by teddykoker — APPROVED** [review]

File: docs/source/reference/object_detection.rst (hunk 3 above)
Comment (kaushikb11) [reply docs/source/reference/object_detection.rst:71]: Added!

**Review by Borda — APPROVED** [review]

File: docs/source/reference/object_detection.rst [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+
+------
+
+*********
+Inference
+*********
+
+The :class:`~flash.vision.ObjectDetector` is already pre-trained on `COCO train2017 <https://cocodataset.org/>`_, a dataset with `91 classes <https://cocodataset.org/#explore>`_ (123,287 images, 886,284 instances).
+
+.. code-block:: json
```
Comment (Borda) [inline docs/source/reference/object_detection.rst:22]:
```suggestion
.. code-block::
```

**Pull request merged**